scripts/github-actions/generate.sh

```diff
@@ -176,8 +176,25 @@ strategy="$(
 					),
 					"git clone --depth 1 https://github.com/docker-library/official-images.git -b master ~/oi",
 
-					# https://github.com/docker-library/bashbrew/pull/43
-					"echo BASHBREW_BUILDKIT_SYNTAX=\"$(< ~/oi/.bashbrew-buildkit-syntax)\" >> \"$GITHUB_ENV\"",
+					(
+						"# https://github.com/docker-library/bashbrew/pull/43",
+						if ([ .meta.entries[].builder ] | index("buildkit")) then
+							"export BASHBREW_ARCH=\(if .os | startswith("windows-") then "windows-" else "" end)amd64",
+							"printf \"BASHBREW_ARCH=%q\\n\" \"$BASHBREW_ARCH\" >> \"$GITHUB_ENV\"",
+							"if [ -x ~/oi/.bin/bashbrew-buildkit-envs.sh ]; then",
+							"\t# https://github.com/docker-library/official-images/pull/14212",
+							"\tbuildkitEnvs=\"$(~/oi/.bin/bashbrew-buildkit-envs.sh)\"",
+							"\teval \"$buildkitEnvs\"",
+							"\tsed <<<\"$buildkitEnvs\" -re \"s/^export[[:space:]]+//\" >> \"$GITHUB_ENV\"",
+							"else",
+							"\tBASHBREW_BUILDKIT_SYNTAX=\"$(< ~/oi/.bashbrew-buildkit-syntax)\"; export BASHBREW_BUILDKIT_SYNTAX",
+							"\tprintf \"BASHBREW_BUILDKIT_SYNTAX=%q\\n\" \"$BASHBREW_BUILDKIT_SYNTAX\" >> \"$GITHUB_ENV\"",
+							"fi",
+							empty
+						else
+							empty
+						end
+					),
```

Comment:
Since this is a little bit hard to read this way, here's the raw version of the final `prepare` string this generates:

```bash
git clone --depth 1 https://github.com/docker-library/official-images.git -b master ~/oi
# https://github.com/docker-library/bashbrew/pull/43
export BASHBREW_ARCH=amd64
printf "BASHBREW_ARCH=%q\n" "$BASHBREW_ARCH" >> "$GITHUB_ENV"
if [ -x ~/oi/.bin/bashbrew-buildkit-envs.sh ]; then
	# https://github.com/docker-library/official-images/pull/14212
	buildkitEnvs="$(~/oi/.bin/bashbrew-buildkit-envs.sh)"
	eval "$buildkitEnvs"
	sed <<<"$buildkitEnvs" -re "s/^export[[:space:]]+//" >> "$GITHUB_ENV"
else
	BASHBREW_BUILDKIT_SYNTAX="$(< ~/oi/.bashbrew-buildkit-syntax)"; export BASHBREW_BUILDKIT_SYNTAX
	printf "BASHBREW_BUILDKIT_SYNTAX=%q\n" "$BASHBREW_BUILDKIT_SYNTAX" >> "$GITHUB_ENV"
fi
# create a dummy empty image/layer so we can --filter since= later to get a meaningful image list
{ echo FROM busybox:latest; echo RUN :; } | docker build --no-cache --tag image-list-marker -
```